### ISSUE TITLE
Test failure: baseservices\\threading\\interlocked\\add\\InterlockedAddInt_2\\InterlockedAddInt_2.cmd

### ISSUE THREAD

**Issue by v-haren** [dotnet/runtime#44139]
failed in job: [runtime-coreclr ilasm 20201031.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=872283&view=ms.vss-test-web.build-test-results-tab&runId=27909720&resultId=105217&paneView=debug)

CoreCLR Windows_NT arm64 Checked ilasmroundtrip @ Windows.10.Arm64v8.Open

Error message
~~~
Return code:      1
Raw output file:      D:\h\w\B2680950\w\B0410974\e\baseservices\threading\Reports\baseservices.threading\interlocked\add\InterlockedAddInt_2\InterlockedAddInt_2.output.txt
Raw output:
BEGIN EXECUTION
 cmd /c ..\InterlockedAddInt\InterlockedAddInt.cmd
BEGIN EXECUTION
D:\h\w\B2680950\p\ildasm.exe /raweh /out=InterlockedAddInt.dasm.il InterlockedAddInt.dll
D:\h\w\B2680950\p\ilasm.exe /output=InterlockedAddInt.asm.dll  InterlockedAddInt.dasm.il

Microsoft (R) .NET IL Assembler.  Version 6.0.0-ci
Copyright (c) Microsoft Corporation.  All rights reserved.
Assembling 'InterlockedAddInt.dasm.il'  to EXE --> 'InterlockedAddInt.asm.dll'
Could not open InterlockedAddInt.dasm.il

***** FAILURE ***** 
EXECUTION OF ILASM - FAILED 1
Expected: 0
Actual: 1
END EXECUTION - FAILED
FAILED
Test Harness Exitcode is : 1
To run the test:
> set CORE_ROOT=D:\h\w\B2680950\p
> D:\h\w\B2680950\w\B0410974\e\baseservices\threading\interlocked\add\InterlockedAddInt_2\InterlockedAddInt_2.cmd
Expected: True
Actual:   False


Stack trace
   at baseservices_threading._interlocked_add_InterlockedAddInt_2_InterlockedAddInt_2_._interlocked_add_InterlockedAddInt_2_InterlockedAddInt_2_cmd() in F:\workspace\_work\1\s\artifacts\tests\coreclr\Windows_NT.arm64.Checked\TestWrappers\baseservices.threading\baseservices.threading.XUnitWrapper.cs:line 18985
~~~

**Comment by JulieLeeMSFT** [comment]
CC @briansull

**Comment by briansull** [comment]
We have a lot of similar issues:

@BruceForstall

Most likely some kind of infrastructure issue due to file locking.

**Comment by BruceForstall** [comment]
Seems like we should add some more checking/retry logic. E.g., make sure the .dasm.il file exists before assembling, then retrying once or twice (after waiting) if it fails with a "Could not open" error. Maybe ildasm crashed and it actually doesn't exist?

**Comment by briansull** [comment]
@BruceForstall Do you on anyone else understand how this IlAsm/IlDasm roundtrip testing is supposed to work?

**Comment by BruceForstall** [comment]
fyi, that logic comes from https://github.com/dotnet/runtime/blob/6553a49e1b501cd3c40ef64999475fecad1c9f19/src/tests/Common/CLRTest.Execute.Batch.targets

I was thinking we add something like:

```
REM IlasmRoundTrip Script
REM Disable Ilasm round-tripping for Linker tests.
REM Todo: Ilasm round-trip on linked binaries.
IF NOT DEFINED DoLink (
  IF DEFINED RunningIlasmRoundTrip (
    ECHO %CORE_ROOT%\ildasm.exe /raweh /out=ThreadStartGenerics.dasm.il ThreadStartGenerics.dll
    %CORE_ROOT%\ildasm.exe /raweh /out=ThreadStartGenerics.dasm.il ThreadStartGenerics.dll
    IF NOT "!ERRORLEVEL!"=="0" (
      ECHO EXECUTION OF ILDASM - FAILED !ERRORLEVEL!
      Exit /b 1
    )
>>>> new
    IF NOT exist ThreadStartGenerics.dasm.il (
      ECHO EXECUTION OF ILDASM - FAILED ThreadStartGenerics.dasm.il is missing
      Exit /b 1
    )

   set ilasm_count=0
:start_ilasm
<<<<
    ECHO %CORE_ROOT%\ilasm.exe /output=ThreadStartGenerics.asm.dll  ThreadStartGenerics.dasm.il
    %CORE_ROOT%\ilasm.exe /output=ThreadStartGenerics.asm.dll  ThreadStartGenerics.dasm.il
    IF NOT "!ERRORLEVEL!"=="0" (
      ECHO EXECUTION OF ILASM - FAILED !ERRORLEVEL!
>>>> new
      if !ilasm_count! LEQ 3 (
        ECHO Trying again
       set /A ilasm_count=ilasm_count +1
       goto start_ilasm
      )
<<<<
      Exit /b 1
    )
  )
)
```

of course, it has to be implemented for both Linux/bash and Windows/cmd.

**Comment by briansull** [comment]
The issue could also be `IlDasm` failing for out of space or some other reason?

**Comment by briansull** [comment]
it has to be implemented for both Linux/bash and Windows/cmd.

All of the reported failure are on Windows, I'm not sure if we are running IlDasm/IlAsm roundtrip testing on Linux.

**Comment by briansull** [comment]
I built the Priority=1 tests locally and run this test 100 times on my system
But it didn't repro for me on my system.

**Comment by briansull** [comment]
Fixed with #46765
